src/reducers/computeRenderedCols.js

```diff
@@ -136,27 +128,21 @@ export default function computeRenderedCols(state, scrollAnchor) {
  *
  * @param state
  * @param columnsContainer
- * @param columnsGroupsContainer
- * @param columnGroupGetter
- * @returns {{columnsToRender: Array, columnOffsets: {}, columnGroupsToRender: Array, columnGroupOffsets: {}}}
+ * @returns {{columnsToRender: Array, columnOffsets: {}, columnsGroupsContainer: Array, columnGroupsToRender: Array, columnGroupOffsets: {}}}
  */
-function computeRenderedFixedColumnsAndGroups(
-  state,
-  columnsContainer,
-  columnsGroupsContainer,
-  columnGroupGetter
-) {
+function computeRenderedFixedColumnsAndGroups(state, columnsContainer) {
   const tableWidth = state.tableSize.width;
 
   let widthUsed = 0;
   const columnsToRender = [];
   const columnGroupsToRender = [];
   const columnOffsets = {};
   const columnGroupOffsets = {};
-  const columnGroupWidths = {};
+  const columnsGroupsContainer = [];
+  let prevColumnGroupIndex = null;
```

Comment:
Why are we recomputing columnsGroupContainer. This means that we are refetching data from user everyTime this is called.

Reply:
Good catch.
@pradeepnschrodinger  is there any reason that we earlier decide to fetch fixedColumnGroups here instead of index.js/initializeFixedColumnWidthsAndOffsets where we are fetching fixedColumns ??

Reply:
I don't think so.

Moving the logic to a single place within `computeRenderedCols.js` sounds good to me.

Reply:
Not done yet, will do it with follow PR